[ADD] accounting/l10n_co: adding automatic DIAN enablement process (#18026)
odoo/documentation: content/applications/finance/fiscal_localizations/colombia.rst

.. _localization/colombia/master-data:

Automatic DIAN Enablement
-------------------------

Odoo automates the |DIAN| enablement process to become an electronic invoicer using the `DIAN Own Software
<https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/como-puedes-facturar-electronicamente/>`_ solution,
sending the requested number of valid customer invoices, credit notes and debit notes to the DIAN

> certification environment.

Reviewer: ```suggestion
sending the required number of valid customer invoices, credit notes, and debit notes to the DIAN
certification environment.
```

To start this process it is necessary to complete the steps of the :ref:`electronic invoicing main configurations

> <localization/colombia/einvoice-configuration>`.

Reviewer: To be more straightforward.
```suggestion
To start this process, complete the :ref:`main configuration steps for electronic invoicing
<localization/colombia/einvoice-configuration>`.
```

.. tip::
   Always run the enablement process in a **test database**. This will also serve as the environment

>    for continued electronic invoicing testing before going to production.

Reviewer: ```suggestion
   Always run the enablement process in a **test database** to serve as the environment for
   continued electronic invoicing testing before going to production.
```

To begin the process, go to :menuselection:`Accounting --> Configuration --> Settings` and complete the

> following steps:

Reviewer: ```suggestion
To begin the process, go to :menuselection:`Accounting --> Configuration --> Settings`, scroll to
the :guilabel:`Colombian Electronic Invoicing` section, and complete the following steps:
```

#. Enable the :guilabel:`Test Environment` checkbox.
#. Enable the :guilabel:`Activate Certification Process` checkbox.
#. Enter the number of documents requested by the |DIAN| for each type of document

> #. Click :guilabel:`Start Enablement Process`.

Reviewer: I was only able to test it on a 19.1 runbot, but isn't it "Begin Certification Process"?

(If not, it will need an update in the 19.1 forward-port.)

Author: Yes it is "Begin Certification Process", seem I paraphrased for some reason :sweat_smile:

.. note::
   The exact number of documents required by the |DIAN| is shown in the portal when you register

>    your operation mode.

Reviewer: ```suggestion
   the operation mode.
```

Odoo will automatically generate the necessary journals, products, and the required documents, which are
submitted to the |DIAN|'s certification environment.

When the process is complete, click :guilabel:`Open` to review the generated documents. You can check each

> document's validation status under its :guilabel:`DIAN` tab.

Reviewer: ```suggestion
When the process is complete, click :guilabel:`Open` to review the generated documents and check
each document's validation status under its :guilabel:`DIAN` tab.
```
```

.. important::
   Confirm in the |DIAN| portal that the company status has changed to **Enabled** before
   proceeding to production. 
   - If this is the case, the process has been finished successfully.

>    - If the company has not been **Enabled**, the process can be repeated until successfully getting enabled.

Reviewer: ```suggestion
   Confirm that the company status has changed to :guilable:`Enabled` in the |DIAN| portal before
   proceeding to production.
   - If the status is :guilable:`Enabled`, the process has been completed successfully.
   - If the company is not :guilable:`Enabled`, repeat the process until it is successfully enabled.
```

.. seealso::

>    - For a demonstration of the process, review the following `video <https://www.youtube.com/watch?v=_0j9kqL9yOU>`_

Reviewer: ```suggestion
   - Watch this `video <https://www.youtube.com/watch?v=_0j9kqL9yOU>`_ for a demonstration of the
   process.
```

.. _localization/colombia/master-data:

Automatic DIAN Enablement
-------------------------

Odoo automates the |DIAN| enablement process to become an electronic invoicer using the `DIAN Own Software
<https://micrositios.dian.gov.co/sistema-de-facturacion-electronica/como-puedes-facturar-electronicamente/>`_ solution,
sending the required number of valid customer invoices, credit notes, and debit notes to the DIAN
certification environment.

To start this process, complete the :ref:`main configuration steps for electronic invoicing
<localization/colombia/einvoice-configuration>`.

.. tip::
   Always run the enablement process in a **test database** to serve as the environment for
   continued electronic invoicing testing before going to production.

To begin the process, go to :menuselection:`Accounting --> Configuration --> Settings`, scroll to
the :guilabel:`Colombian Electronic Invoicing` section, and complete the following steps:

#. Enable the :guilabel:`Test Environment` checkbox.
#. Enable the :guilabel:`Activate Certification Process` checkbox.
#. Enter the number of documents requested by the |DIAN| for each type of document

> #. Click :guilabel:`Start Enablement Process`.

Author: ```suggestion
#. Click :guilabel:`Begin Certification Process`.
```

> #. Click :guilabel:`Begin Certification Process`.

Reviewer: I'm not sure, but I think we should save.
```suggestion
#. Click :guilabel:`Begin Certification Process`, then :guilabel:`Save`.
```

.. note::
   The exact number of documents required by the |DIAN| is shown in the portal when you register
   the operation mode.

Odoo will automatically generate the necessary journals, products, and the required documents, which are

> submitted to the |DIAN|'s certification environment.

Reviewer: I think it's more straightforward this way.
```suggestion
The necessary journals, products, and required documents are automatically generated and submitted
to the |DIAN|'s certification environment.
```

When the process is complete, click :guilabel:`Open` to review the generated documents and check
each document's validation status under its :guilabel:`DIAN` tab.

.. important::
   Confirm that the company status has changed to :guilable:`Enabled` in the |DIAN| portal before
   proceeding to production.
   - If the status is :guilable:`Enabled`, the process has been completed successfully.

>    - If the company is not :guilable:`Enabled`, repeat the process until it is successfully enabled.

Reviewer: My bad, sorry about that ☺️ 
```suggestion
   Confirm that the company status has changed to :guilabel:`Enabled` in the |DIAN| portal before
   proceeding to production.
   - If the status is :guilabel:`Enabled`, the process has been completed successfully.
   - If the company is not :guilabel:`Enabled`, repeat the process until it is successfully enabled.
```
